js/server_functions.js

```diff
@@ -0,0 +1,127 @@
+const fetch = require("./fetch");
+const fs = require("fs");
+const path = require("path");
+const Log = require("logger");
+
+/**
+ * Gets the config.
+ *
+ * @param {Request} req - the request
+ * @param {Response} res - the result
+ */
+function getConfig(req, res) {
+	res.send(config);
+}
+
+/**
+ * A method that forewards HTTP Get-methods to the internet to avoid CORS-errors.
+ *
+ * Example input request url: /cors?sendheaders=header1:value1,header2:value2&expectedheaders=header1,header2&url=http://www.test.com/path?param1=value1
+ *
+ * Only the url-param of the input request url is required.
+ *
+ * @param {Request} req - the request
+ * @param {Response} res - the result
+ */
+async function cors(req, res) {
+	try {
+		const urlRegEx = "url=(.+?)$";
+		let url = "";
+
+		let match = new RegExp(urlRegEx, "g").exec(req.url);
+		if (!match) {
+			url = "invalid url: " + req.url;
+			Log.error(url);
+			res.send(url);
+		} else {
+			url = match[1];
+
+			const headersToSend = GetHeadersToSend(req.url);
+			const expectedRecievedHeaders = GeExpectedRecievedHeaders(req.url);
+
+			Log.log("cors url: " + url);
+			const response = await fetch(url, { headers: headersToSend });
+
+			for (const header of expectedRecievedHeaders) {
+				const headerValue = response.headers.get(header);
+				if (header) res.set(header, headerValue);
+			}
+			const data = await response.text();
+			res.send(data);
+		}
+	} catch (error) {
+		Log.error(error);
+		res.send(error);
+	}
+}
+
+/**
+ * Gets headers and values to attatch to the web request.
+ *
+ * @param {string} url - The url containing the headers and values to send.
+ * @returns {object} An object specifying name and value of the headers.
+ */
+function GetHeadersToSend(url) {
```

Comment:
Capitalized function names? I guess to mark them somehow as private? Or what is the reasoning here?

Reply:
Whoops, had been writing C# all day. I'll fix it together with the changes requested below 👍